Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp

```diff
@@ -291,50 +296,95 @@ void Wiimote::WriteData(const wm_write_data* const wd)
     PanicAlert("WriteData: unimplemented parameters!");
     break;
   }
+
+  SendAck(RT_WRITE_DATA, error_code);
 }
 
 /* Read data from Wiimote and Extensions registers. */
 void Wiimote::ReadData(const wm_read_data* const rd)
 {
-  u16 address = Common::swap16(rd->address);
-  u16 size = Common::swap16(rd->size);
+  if (m_read_request.size)
+  {
+    // There is already an active read request.
+    // a real wiimote ignores the new one.
+    return;
+  }
+
+  // Save the request and process it on the next "Update()" calls
+  m_read_request.space = rd->space;
+  m_read_request.slave_address = rd->slave_address;
+  m_read_request.address = Common::swap16(rd->address);
+  // A zero size request is just ignored, like on the real wiimote.
+  m_read_request.size = Common::swap16(rd->size);
+
+  INFO_LOG(WIIMOTE, "Wiimote::ReadData: %d @ 0x%02x @ 0x%02x (%d)", m_read_request.space,
+          m_read_request.slave_address, m_read_request.address, m_read_request.size);
+
+  // Send up to one read-data-reply.
+  // If more data needs to be sent it will happen on the next "Update()"
+  ProcessReadDataRequest();
+}
+
+bool Wiimote::ProcessReadDataRequest()
+{
+  // Limit the amt to 16 bytes
+  // AyuanX: the MTU is 640B though... what a waste!
+  u16 const bytes_to_read = std::min((u16)16, m_read_request.size);
+
+  if (0 == bytes_to_read)
+  {
+    // No active request:
+    return false;
+  }
 
-  //INFO_LOG(WIIMOTE, "Wiimote::ReadData: %d @ 0x%02x @ 0x%02x (%d)", rd->space, rd->slave_address, address, size);
+  u8 data[23] = {};
+  data[0] = 0xA1;
```

Comment:
Magic number?

Reply:
That is the hex for an input read.  A1 for Input, A2 for output.

https://wiibrew.org/wiki/Wiimote#HID_Interface

Reply:
Magic number has been killed.